src/components/PostDetails.tsx

```diff
@@ -1,118 +1,162 @@
-import React from 'react';
+import {
+  FC, memo, useCallback, useEffect, useState,
+} from 'react';
+import {
+  addComment, deleteComment, getCommentsByPostId,
+} from '../api/comments';
+import { Comment, CommentData } from '../types/Comment';
+import { Post } from '../types/Post';
 import { Loader } from './Loader';
 import { NewCommentForm } from './NewCommentForm';
 
-export const PostDetails: React.FC = () => {
+type Props = {
+  post: Post
+};
+
+export const PostDetails: FC<Props> = memo(({ post }) => {
+  const { id: postId, title, body: postBody } = post;
+
+  const [isLoading, setIsLoading] = useState(false);
+  const [comments, setComments] = useState<Comment[]>([]);
+  const [isCommentWriting, setIsCommentWriting] = useState(false);
+  const [errorMessage, setErrorMessage] = useState('');
+
+  const handleCommentDelete = async (commentId: number) => {
+    setErrorMessage('');
+
+    const currentComments = comments;
+
+    try {
+      setComments(comments.filter(({ id }) => id !== commentId));
+
+      await deleteComment(commentId);
+    } catch {
+      setErrorMessage("Can't delete comment");
+      setComments(currentComments);
+    }
+  };
+
+  const handleCommentAdd = useCallback(async (commentData: CommentData) => {
+    try {
+      setErrorMessage('');
+
+      const uploadedComment = await addComment({ postId, ...commentData });
+
+      setComments(prevComments => [...prevComments, uploadedComment]);
+
+      return true;
```

Comment:
do you need this return?

Reply:
yes, i use returned value in NewComentForm

i clear body field only when comment was added successfully
and keep it, if we got error

<img width="295" alt="image" src="https://user-images.githubusercontent.com/83500664/217322433-52b58de6-00da-4c89-bfb1-49c63cd0b796.png">